pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -438,28 +438,10 @@ func (p *staticPolicy) calculateHints(machineState state.NUMANodeMap, pod *v1.Po
 		maskBits := mask.GetBits()
 		singleNUMAHint := len(maskBits) == 1
 
-		// the node already in group with another node, it can not be used for the single NUMA node allocation
-		if singleNUMAHint && len(machineState[maskBits[0]].Cells) > 1 {
-			return
-		}
-
```

Comment:
I see. So basically you move this call below the loop (so we don't break out early), and then repurpose the loop to *only* update `minAffinitySize`(if possible).

Then, if you make it through this loop (with a new `minAffinity` or one equal to a previously calculated `minAffinity`), only then do you go through the process of calculating the set of hints using (basically) the same logic you did before.

I didn't verify in this review that the existing logic is necessarily correct (I assume we did that in a previous PR), but the minor refactoring here seems makes sense.